src/validation/validateStates.ts

```diff
@@ -47,6 +47,30 @@ interface ValidateCatchResult {
     reachedStates: { [ix: string]: boolean }
 }
 
+function validateParameters(parametersPropNode: PropertyASTNode, document: TextDocument): Diagnostic[] {
+    let diagnostics: Diagnostic[] = []
+    const valueNode = parametersPropNode.valueNode
+
+    if (isObjectNode(valueNode)) {
+        valueNode.properties.forEach(prop => {
+            if (prop.keyNode.value.endsWith('.$')) {
+                const propValue = prop.valueNode.value
+
+                if (typeof propValue !== 'string' || !propValue.startsWith('$')) {
```

Comment:
Do we want to trim? Not sure whether the JSON path is still valid if prefixed with spaces.

Reply:
That's a very valid question. I checked in StepFunctions console. We don't need to trim - it always starts with $.